rcl/src/rcl/security.c

```diff
@@ -157,33 +157,33 @@ char * rcl_get_secure_root(
     if (0 == strcmp("", env_buf)) {
       return NULL;  // environment variable was empty
     }
-    ros_secure_directory_override = false;
+    ros_secure_enclave_override = false;
   }
 
   // found a usable environment variable, copy into our memory before overwriting with next lookup
-  char * ros_secure_root_env = rcutils_strdup(env_buf, *allocator);
+  char * ros_secure_keystore_env = rcutils_strdup(env_buf, *allocator);
 
   char * secure_root = NULL;
-  if (ros_secure_directory_override) {
-    secure_root = rcutils_strdup(ros_secure_root_env, *allocator);
+  if (ros_secure_enclave_override) {
+    secure_root = exact_match_lookup(ros_secure_keystore_env, ros_secure_keystore_env, allocator);
```

Comment:
@ivanpauno < actually, this doesn't look quite right. We should store the string from `ROS_SECURITY_ENCLAVE_OVERRIDE` into something like `ros_secure_enclave_override_env` instead of `ros_secure_keystore_env`. The `ros_secure_keystore_env` should be set to the contents of `ROS_SECURITY_KEYSTORE` regardless if `ROS_SECURITY_ENCLAVE_OVERRIDE` is set as well.

Reply:
why?

Reply:
Before, the override was a path, and not just a name.
The change looks good to me, if this is the desired behavior now.

Reply:
As I understand it, the override is now a name/identity within the given keystore, which allows for future keystores that are not filesystem-based.

Reply:
Ok, I think this also matches the behavior change described here https://github.com/ros2/design/pull/274/commits/98200d8e3b273194afc3170f3a9b65eaab689226, which I didn't see before. LGTM!